library/Ivoz/Provider/Domain/Service/HolidayDate/CheckEventDateCollision.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Ivoz\Provider\Domain\Service\HolidayDate;
+
+use Ivoz\Provider\Domain\Model\HolidayDate\HolidayDate;
+use Ivoz\Provider\Domain\Model\HolidayDate\HolidayDateDto;
+use Ivoz\Provider\Domain\Model\HolidayDate\HolidayDateInterface;
+
+/**
+ * Class CheckEventDateCollision
+ * @package Ivoz\Provider\Domain\Service\HolidayDate
+ */
+class CheckEventDateCollision implements HolidayDateLifecycleEventHandlerInterface
+{
+    const PRE_PERSIST_PRIORITY = self::PRIORITY_NORMAL;
+
+    public function __construct()
+    {
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            self::EVENT_PRE_PERSIST => self::PRE_PERSIST_PRIORITY,
+        ];
+    }
+
+    public function execute(HolidayDateInterface $holidayDate)
+    {
+        $calendar = $holidayDate->getCalendar();
+        $calendarHolidays = $calendar->getHolidayDates();
```

Comment:
What about putting here some criteria to avoid self collision and equal event date checks?

Reply:
Something like:


```        $eventsCriteria = CriteriaHelper::fromArray(
            [
                array('id', 'ne', $holidayDate->getId()),
                array('evenDate', 'eq', $holidayDate->getEventDate()),
            ]
        );

        $calendar = $holidayDate->getCalendar();
        $calendarHolidays = $calendar->getHolidayDates($eventsCriteria);
```

Reply:
Yes :)